### PR TITLE
update spec for buckets function

### PR DESCRIPTION
**Pull request by aanthony1243**
closes https://github.com/influxdata/platform/issues/769

### CODE REVIEW

File: docs/SPEC.md [hunk 1]
```diff
@@ -1198,6 +1198,22 @@ Example:
...
+#### Buckets
+
+Buckets is a type of data source that retrieves a list of buckets that the caller is authorized to access.  
+It takes no input parameters and produces an output table with the following columns: 
+
+* name: the name of the bucket
+* id: the internal ID of the bucket
+* organization: the organization this bucket belongs to
+* organizationID: the internal ID of the organization
+* Retention Policy: the name of the retention policy, if present
```
Comment (nathanielc) [inline docs/SPEC.md]: Do retention policies have names? I didn't think they did.

File: docs/SPEC.md [hunk 2]
```diff
@@ -1198,6 +1198,22 @@ Example:
...
+* RetentionPeriod: the duration of time for which data is held in this bucket
```
Comment (nathanielc) [inline docs/SPEC.md]: These cols look good, could you add the data type of the column in the description somewhere?

File: docs/SPEC.md [hunk 3]
```diff
@@ -1198,6 +1198,22 @@ Example:
...
+
+Example: 
+
+    buckets() |> filter(fn: (r) => r["organization"] == "my-org")
```
Comment (nathanielc) [inline docs/SPEC.md]: nit, could you use the `r.organization` syntax? I prefer it for the examples as it is less noisy.

File: docs/SPEC.md (hunk 1 above)
Comment (aanthony1243) [reply docs/SPEC.md]: in what context?  in v1.X they do: 

> show retention policies
name    duration shardGroupDuration replicaN default
----    -------- ------------------ -------- -------
autogen 0s       168h0m0s           1        true

**Pull request merged**